### DISCUSSION TITLE
Dialogue to set proxy for Tor connection

### DISCUSSION THREAD

**Discussion by w0000000t** [bisq-network/bisq#6025] [Ideas]
Just now I followed an issue at Matrix support room, where a user from China singlehandedly solved their own problem to overcome the Great Firewall.
They are running a proxy on their own PC to connect to the internet, bridges didn't work from Bisq, and they were looking for a way to have Bisq use their own proxy to then hop on Tor.
They got the usual "Bisq stuck at 1/4" issue.

They finally tracked the `torrc` file inside `desktop-1.8.2-all.jar`, edited adding `Socks5Proxy 127.0.0.1:1080` at the end, and replacing the file in the archive.

This let them use Bisq successfully.

Maybe this can be added to Settings > Network?

**Comment by w0000000t** [comment]
Created a guide on the wiki:
https://bisq.wiki/Running_Bisq_in_China

**Reply by chimp1984** [reply]
Torrc config should be supported as program argument. But not familiar with the details but I am pretty sure it was at least working at some time, and likely is still working, would just require a dev to document it.

**Reply by w0000000t** [reply]
definitely easier to have it available on commandline rather than editing the jar, until it is confirmed users can follow the nerd way 😉

**Reply by ripcurlx** [reply]
At least in the codebase it says, that there is an option for `torrcFile` and `torrcOptions`
